src/analysis/useComputeImages.ts

```diff
@@ -0,0 +1,63 @@
+import { LoadedState, withHandlers } from '@terra-ui-packages/core-utils';
+import { useEffect, useState } from 'react';
+import { ComputeImageProvider } from 'src/libs/ajax/compute-image-providers/ComputeImageProvider';
+import { withErrorReporting } from 'src/libs/error';
+import { useCancellation, useStore } from 'src/libs/react-utils';
+import { workspaceStore } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+/**
+ * Refers to a docker image of a Terra VM.
+ * Sample response:
+     {
+        "id": "terra-jupyter-bioconductor",
+        "image": "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.2.1",
+        "label": "R / Bioconductor: (Python 3.10.11, R 4.3.1, Bioconductor 3.17, tidyverse 2.0.0)",
+        "packages": "https://storage.googleapis.com/terra-docker-image-documentation/terra-jupyter-bioconductor-2.2.1-versions.json",
+        "requiresSpark": false,
+        "updated": "2023-07-13",
+        "version": "2.2.1"
+    },
+ */
+export interface ComputeImage {
+  id: string;
+  image: string;
+  isCommunity?: boolean;
+  isRStudio?: boolean;
+  label: string;
+  packages: string;
+  requiresSpark: boolean;
+  updated: string;
+  version: string;
+}
+
+export interface ComputeImageStore {
+  refreshStore: () => Promise<void>;
+  loadedState: LoadedState<ComputeImage[], unknown>;
+}
+
+export const useComputeImages = (): ComputeImageStore => {
+  const signal = useCancellation();
+  const [loading, setLoading] = useState(false);
+  const [images, setImages] = useState<ComputeImage[]>([]);
+  const workspace: WorkspaceWrapper = useStore(workspaceStore);
```

Comment:
I'm pretty sure useStore can be given a type for type safety similar to useState.... useStore<MyType>.  This is very helpful in general, and specifically with workspace since wrapper or no-wrapper is easily confused.